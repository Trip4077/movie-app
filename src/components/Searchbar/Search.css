/*
---------------------------Searchbar Styles---------------------------
*/

.search {
    text-align: center;
  }
  
  .search h3 {
    margin: 0;
    margin-top: 35px;
  }
  
  .searchbar {
    position: relative;
  
    width: 70%;
    margin: 30px auto;
  }
  
  #search {
    font-size: 24px;
    letter-spacing: 1px;
  }
  
  #searchbar-icon {
    position: absolute;
    top: 12px;
    right: 0;
    z-index: 500;
  
    background-color: transparent;
    border: none;
  }
  
  #searhbar-icon, #searchbar-icon i:hover {
    cursor: pointer;
  }
  
  .no-results {
    text-align: center
  }
  