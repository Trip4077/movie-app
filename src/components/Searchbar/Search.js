import React from 'react';

const Search = props => {
    return(
        <form onSubmit={props.performSearch}>
            <input type='text'
               name='search'
               value={props.search}
               onChange={props.handleChange}
               placeholder='Enter movie title...' />
        </form>
    );
}

export default Search;