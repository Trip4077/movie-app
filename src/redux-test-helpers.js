import React from 'react';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { MemoryRouter } from 'react-router';

import { render } from 'react-testing-library';

import * as ACTIONS from './actions';

function reducer(state = { userReducer: { user: { id: 1 }}}, action) {
    switch(action.type) {
        case ACTIONS.LOADING:
            return {
                ...state,
                loading: true
            }

        case ACTIONS.END_LOAD:
            return {
                ...state,
                loading: false
            }

        case ACTIONS.UPDATE_FAV_LIST:
            return {
                loading: false,
                favorites: action.payload,
                schedule: state.schedule
            }

        case ACTIONS.GET_INFO:
            return {
                ...state,
                loading: false,
                info: action.payload
            }

        case ACTIONS.UPDATE_SCHEDULE:
            return {
                loading: false,
                favorites: state.favorites,
                schedule: action.payload
            }

        case ACTIONS.TEXT_SENT:
            return {
                ...state,
                loading: false
            }

            case ACTIONS.REGISTER: 
            return {
                loading: false,
                userID: action.payload
            }

        case ACTIONS.LOGIN:
            localStorage.setItem( 'token', action.payload.token )

            delete action.payload.user.password

            return {
                loading: false,
                user: action.payload.user
            }

        case ACTIONS.LOGOUT:
            return { ...initialState }

        default:
            return {
                ...state
            }
    }
}

export function renderWithRedux( 
    ui, 
    { initialState, store = createStore(reducer, initialState) } = {} ) {
        return {
            ...render( <Provider store={store}>
                         <MemoryRouter>
                            <div> {ui} </div>
                         </MemoryRouter>
                       </Provider> ),
            store
        }
    };