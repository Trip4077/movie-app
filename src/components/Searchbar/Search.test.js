import React from 'react';

import { cleanup, fireEvent } from 'react-testing-library';
import { renderWithRedux } from '../../redux-test-helpers';

import bananna from 'react-test-renderer';

import Search from './Search';

afterEach(cleanup);

describe('<Search />', () => {

  it('renders without crashing', () => {
    renderWithRedux(<Search />);
  });

  it('renders a search form', () => {
    const { getByTestId } = renderWithRedux(<Search />);

    getByTestId('search-form');
  })

  it('renders "Search for Title" label', () => {
    const { getByLabelText } = renderWithRedux(<Search />);

    getByLabelText(/search by title/i);
  });

  it('renders search button', () => {
    const { getByText } = renderWithRedux(<Search />);

    getByText(/search$/i);
  });

  it('renders passed value as input value', () => {
    const { getByDisplayValue } = renderWithRedux(<Search search='movie' />);

    getByDisplayValue(/movie/i);
  });

  it('fires props function when search button is clicked', () => {
    const mock = jest.fn();
    const { getByText } = renderWithRedux(<Search performSearch={mock} />);

    const button = getByText(/search$/i).parentElement;

    fireEvent.click(button);

    expect(mock).toHaveBeenCalled();
  });

  it('fires props function when form is submitted', () => {
    const mock = jest.fn();
    const { getByTestId } = renderWithRedux(<Search performSearch={mock} />);

    const form = getByTestId('search-form');

    fireEvent.submit(form, {});

    expect(mock).toHaveBeenCalled();
  });

  it('fires props function when input is changed', () => {
    const mock = jest.fn();
    const { getByDisplayValue } = renderWithRedux(<Search search='movie' handleChange={mock} />);

    const input = getByDisplayValue(/movie/i);
    
    fireEvent.change(input, { target: { value: 'movies' } });

    expect(mock).toHaveBeenCalled();
  });

  it('matches snapshot - default', () => {
    const tree = bananna.create(<Search />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot - with props', () => {
    const tree = bananna.create(<Search search={'some value'} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});