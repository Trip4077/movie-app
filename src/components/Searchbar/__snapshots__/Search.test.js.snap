// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches snapshot - default 1`] = `
<form
  data-testid="search-form"
>
  <div
    className="input-field searchbar"
  >
    <input
      id="search"
      name="search"
      type="text"
    />
    <label
      htmlFor="search"
    >
      Search By Title
    </label>
    <button
      id="searchbar-icon"
      type="submit"
    >
      <i
        className="material-icons small"
      >
        search
      </i>
    </button>
  </div>
</form>
`;

exports[`<Search /> matches snapshot - with props 1`] = `
<form
  data-testid="search-form"
>
  <div
    className="input-field searchbar"
  >
    <input
      defaultValue="some value"
      id="search"
      name="search"
      type="text"
    />
    <label
      htmlFor="search"
    >
      Search By Title
    </label>
    <button
      id="searchbar-icon"
      type="submit"
    >
      <i
        className="material-icons small"
      >
        search
      </i>
    </button>
  </div>
</form>
`;
